core/src/nucleus/ribosome/api/capabilities.rs

```diff
@@ -0,0 +1,83 @@
+use crate::{
+    agent::actions::commit::commit_entry,
+    nucleus::ribosome::{api::ZomeApiResult, Runtime},
+};
+use holochain_core_types::{
+    cas::content::Address,
+    entry::{cap_entries::CapTokenGrant, Entry},
+    error::HolochainError,
+};
+use holochain_wasm_utils::api_serialization::capabilities::GrantCapabilityArgs;
+use std::convert::TryFrom;
+use wasmi::{RuntimeArgs, RuntimeValue};
+
+pub fn invoke_grant_capability(runtime: &mut Runtime, args: &RuntimeArgs) -> ZomeApiResult {
+    let context = runtime.context()?;
+    // deserialize args
+    let args_str = runtime.load_json_string_from_args(&args);
+    let args = match GrantCapabilityArgs::try_from(args_str.clone()) {
+        Ok(input) => input,
+        Err(..) => return ribosome_error_code!(ArgumentDeserializationFailed),
```

Comment:
Wouldn't hurt to add a log statement here about details of the deserialization error. Eventually I would even change the ribosome errors to be more structured but that's outside the scope of this PR.